automatic/git.install/tools/helpers.ps1

```diff
@@ -0,0 +1,48 @@
+function Get-InstallKey()
+{
+    $keyName = 'Git_is1'
+    $installKey = if ((Get-ProcessorBits 64) -and $env:chocolateyForceX86 -ne 'true') {
+             "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$keyName"
+    } else { "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\$keyName" }
+
+    $userInstallKey = "HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall\$keyName"
+    if (Test-Path $userInstallKey) { $installKey = $userInstallKey }
+
+    mkdir $installKey -ea 0 | Out-Null
+    $installKey
+}
+
+function Set-InstallerRegistrySettings( [HashTable]$pp )
+{
+    $installkey = Get-InstallKey
+    $ino = "Inno Setup CodeFile:"
+
+    if ($pp.GitOnlyOnPath)         { New-ItemProperty $InstallKey -Name "$ino Path Option"          -Value "Cmd"            -Force }
+    if ($pp.GitAndUnixToolsOnPath) { New-ItemProperty $InstallKey -Name "$ino Path Option"          -Value "CmdTools"       -Force }
+    if ($pp.WindowsTerminal)       { New-ItemProperty $InstallKey -Name "$ino Bash Terminal Option" -Value "ConHost"        -Force }
+    if ($pp.NoAutoCrlf)            { New-ItemProperty $InstallKey -Name "$ino CRLF Option"          -Value "CRLFCommitAsIs" -Force }
+}
+
+function Get-InstallComponents( [HashTable]$pp )
+{
+    $res = "icons", "assoc", "assoc_sh"
+
+    if ($pp.NoShellIntegration) {
+        Write-Host "Parameter: no git shell integration"
+    } else { $res += "ext", "ext\shellhere", "ext\guihere" }
+
+    # Make our install work properly when running under SYSTEM account (Chef Cliet Service, Puppet Service, etc)
+    $isSystem = ([System.Security.Principal.WindowsIdentity]::GetCurrent()).IsSystem
+    if ( !$isSystem ) { $res += "icons\quicklaunch" }
+
+    if ($res.Length -eq 0) { return }
+    '/COMPONENTS="{0}"' -f ($res -join ",")
```

Comment:
Why so much appoligies ? 😕 

Lets analyze:

```powershell
function Get-InstallComponents( [HashTable]$pp )
{
    $res = "icons", "assoc", "assoc_sh"   ; #1

    if ($pp.NoShellIntegration) {  #2
        Write-Host "Parameter: no git shell integration"
    } else { $res += "ext", "ext\shellhere", "ext\guihere" }

    # Make our install work properly when running under SYSTEM account (Chef Cliet Service, Puppet Service, etc)
    $isSystem = ([System.Security.Principal.WindowsIdentity]::GetCurrent()).IsSystem
    if ( !$isSystem ) { $res += "icons\quicklaunch" } #3

    if ($res.Length -eq 0) { return }
    '/COMPONENTS="{0}"' -f ($res -join ",") 
}
```

- 1 means those opotions are ALWYAS applied
- 2 and 3 are contextual. So you could look your proposed version of the code and espect that changing default there would be enough, but it wouldn't. This code states that intention.
- Above code works as a blackbox and you don't have to read anything else in the package to understand it or change it. Contary to that, you need to be aware or 100+ lines of code to change those 7 lines.